File: TelerikUIExamplesInCSharp/Examples/SideDrawer/GettingStartedExample/SideDrawerGettingStarted.cs
using System;
using UIKit;
using Foundation;
using CoreGraphics;
using ObjCRuntime;
using TelerikUI;

namespace Examples
{
	public class SideDrawerGettingStarted : ExampleViewController
	{
		private TKSideDrawerSection primarySection;
		private TKSideDrawerSection labelsSection;
		SideDrawerDelegate sideDrawerDelegate;

		public UINavigationItem NavItem {
			get;
			set;
		}

		public TKSideDrawerView SideDrawerView {
			get;
			set;
		}

		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();
			this.NavigationController.InteractivePopGestureRecognizer.Enabled = false;
		
			this.SideDrawerView = new TKSideDrawerView (this.View.Bounds);
			this.SideDrawerView.AutoresizingMask = UIViewAutoresizing.FlexibleHeight | UIViewAutoresizing.FlexibleWidth;
			this.View.AddSubview (this.SideDrawerView);

			UIImageView backgroundView = new UIImageView (this.SideDrawerView.MainView.Bounds);
			backgroundView.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
			backgroundView.Image = new UIImage ("sdk-examples-bg.png");
			this.SideDrawerView.MainView.AddSubview (backgroundView);

			UINavigationBar navBar = new UINavigationBar (new CGRect (0, 0, this.SideDrawerView.MainView.Bounds.Width, 44));
			navBar.AutoresizingMask = UIViewAutoresizing.FlexibleWidth;
			this.NavItem = new UINavigationItem ();
			UIBarButtonItem showSideDrawer = new UIBarButtonItem (new UIImage ("menu.png"), UIBarButtonItemStyle.Plain, this, new Selector ("ShowSideDrawer"));
			this.NavItem.LeftBarButtonItem = showSideDrawer;
			navBar.Items = new UINavigationItem[] { this.NavItem };
			this.SideDrawerView.MainView.AddSubview (navBar);

			primarySection = new TKSideDrawerSection ("Primary");
			primarySection.AddItem ("Social");
			primarySection.AddItem ("Promotions");

			labelsSection = new TKSideDrawerSection ("Labels");
			labelsSection.AddItem ("Important");
			labelsSection.AddItem ("Starred");
			labelsSection.AddItem ("Sent Mail");
			labelsSection.AddItem ("Drafts");

			this.sideDrawerDelegate = new SideDrawerDelegate ();
			TKSideDrawer sideDrawer = this.SideDrawerView.SideDrawer;
			sideDrawer.HeaderView = new SideDrawerHeader (true, this, new Selector ("DismissSideDrawer"));
			sideDrawer.AddSection (primarySection);
			sideDrawer.AddSection (labelsSection);
			sideDrawer.Delegate = this.sideDrawerDelegate;
			sideDrawer.Style.HeaderHeight = 44;
		}

		public override void ViewWillDisappear (bool animated)
		{
			base.ViewWillDisappear (animated);
			this.NavigationController.InteractivePopGestureRecognizer.Enabled = true;
		}
			
		[Export ("ShowSideDrawer")]
		public void ShowSideDrawer()
		{
			this.SideDrawerView.SideDrawer.Show();
		}

		[Export ("DismissSideDrawer")]
		public void DismissSideDrawer()
		{
			this.SideDrawerView.SideDrawer.Dismiss();
		}

		class SideDrawerDelegate : TKSideDrawerDelegate
		{
			public override void UpdateVisualsForSection (TKSideDrawer sideDrawer, int sectionIndex)
			{
				TKSideDrawerSection section = sideDrawer.Sections[sectionIndex];
				section.Style.ContentInsets = new UIEdgeInsets (0, -15, 0, 0);
			}

			public override void UpdateVisualsForItem (TKSideDrawer sideDrawer, NSIndexPath indexPath)
			{
				TKSideDrawerItem item = sideDrawer.Sections[indexPath.Section].Items[indexPath.Item];
				item.Style.ContentInsets = new UIEdgeInsets (0, -5, 0, 0);
				item.Style.SeparatorColor = new TKSolidFill (UIColor.Clear);
			}
		}
	}
}

File: TelerikUIExamplesInSwift2.0/TelerikUIExamplesInSwift/SideDrawerGettingStarted.swift
//
//  SideDrawerGettingStarted.swift
//  TelerikUIExamplesInSwift
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//

class SideDrawerGettingStarted: ExampleViewController, TKSideDrawerDelegate {

    let sideDrawerView = TKSideDrawerView()
    let navItem = UINavigationItem()
    
    override func viewDidLoad() {
        super.viewDidLoad()

        self.navigationController!.interactivePopGestureRecognizer!.enabled = false

        self.sideDrawerView.frame = self.exampleBounds
        self.view.addSubview(sideDrawerView)
        
        let mainView = sideDrawerView.mainView
        
        let backgroundView = UIImageView(frame: mainView.bounds)
        backgroundView.autoresizingMask = UIViewAutoresizing(rawValue:UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        backgroundView.image = UIImage(named: "sdk-examples-bg")
        mainView.addSubview(backgroundView)
        
        let navigationBar = UINavigationBar(frame: CGRectMake(0, 0, mainView.bounds.size.width, 44))
        navigationBar.autoresizingMask = UIViewAutoresizing.FlexibleWidth
        let showSideDrawerButton = UIBarButtonItem(image: UIImage(named: "menu"), style: UIBarButtonItemStyle.Plain, target: self, action: "showSideDrawer")
        navItem.leftBarButtonItem = showSideDrawerButton
        navigationBar.items = [navItem]
        mainView.addSubview(navigationBar)
        
        let sideDrawer = sideDrawerView.sideDrawer
        sideDrawer.delegate = self
        sideDrawer.style.headerHeight = 44
        sideDrawer.headerView = SideDrawerHeaderView(addButton: true, target: self, selector: Selector("dismissSideDrawer"))
        
        var section = sideDrawer.addSectionWithTitle("Primary")
        section.addItemWithTitle("Social")
        section.addItemWithTitle("Promotions")
        
        section = sideDrawer.addSectionWithTitle("Label")
        section.addItemWithTitle("Important")
        section.addItemWithTitle("Starred")
        section.addItemWithTitle("Sent Mail")
        section.addItemWithTitle("Drafts")
    }
    
    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        self.sideDrawerView.frame = self.exampleBounds
    }
    
    override func viewWillDisappear(animated: Bool) {
        super.viewWillDisappear(animated)
        
        if (self.navigationController!.respondsToSelector("interactivePopGestureRecognizer")) {
            self.navigationController!.interactivePopGestureRecognizer!.enabled = true
        }
    }
    
    
    func showSideDrawer() {
        sideDrawerView.sideDrawer.show()
    }
    
    func dismissSideDrawer() {
        sideDrawerView.sideDrawer.dismiss()
    }
    
    func sideDrawer(sideDrawer: TKSideDrawer!, updateVisualsForSection sectionIndex: Int) {
        let section = sideDrawer.sections()[sectionIndex] as! TKSideDrawerSection
        section.style.contentInsets = UIEdgeInsetsMake(0, -15, 0, 0)
    }
    
    func sideDrawer(sideDrawer: TKSideDrawer!, updateVisualsForItemAtIndexPath indexPath: NSIndexPath!) {
        let currentItem = (sideDrawer.sections()[indexPath.section] as! TKSideDrawerSection).items()[indexPath.item] as! TKSideDrawerItem
        currentItem.style.contentInsets = UIEdgeInsetsMake(0, -10, 0, 0)
        currentItem.style.separatorColor = TKSolidFill(color: UIColor.clearColor())
    }
}
